### PR TITLE
Better handling of strings length

### PR DESCRIPTION
**Pull request by albertodonato**

### CODE REVIEW

File: snap-preload.c [hunk 1]
```diff
@@ -27,14 +27,14 @@ static int DEBUG = 0;
 #define log(FORMAT, ...) if (DEBUG) {fprintf(stderr, "snap-preload: " FORMAT "\n", __VA_ARGS__);}
 
 static char *adjust_shm_path(const char *orig_path) {
-  char *new_path = malloc(PATH_MAX);
+  int path_len = strlen(orig_path) + strlen(SNAP_INSTANCE_NAME) + 8;
```
Comment (alexsander-souza) [inline snap-preload.c]: avoid magic number, use `sizeof("/snap..")` instead

File: snap-preload.c [hunk 2]
```diff
@@ -27,14 +27,14 @@ static int DEBUG = 0;
...
+  char *new_path = malloc(path_len);
   if (SNAP_INSTANCE_NAME) {
     const char *path = (orig_path[0] == '/') ? &(orig_path[1]) : orig_path;
-
-    snprintf(new_path, PATH_MAX, "/snap.%s.%s", SNAP_INSTANCE_NAME, path);
+    snprintf(new_path, path_len , "/snap.%s.%s", SNAP_INSTANCE_NAME, path);
     log("shm path rewritten: %s -> %s", orig_path, new_path);
   } else {
-    new_path = strcpy(new_path, orig_path);
+    new_path = strncpy(new_path, orig_path, strlen(orig_path));
```
Comment (alexsander-souza) [inline snap-preload.c]: use the size of the destination string as limit, unless you want to truncate the string (use `MAX(path_len, strlen(orig_path))` in this case)

File: snap-preload.c [hunk 3]
```diff
@@ -27,14 +27,14 @@ static int DEBUG = 0;
 #define log(FORMAT, ...) if (DEBUG) {fprintf(stderr, "snap-preload: " FORMAT "\n", __VA_ARGS__);}
 
 static char *adjust_shm_path(const char *orig_path) {
-  char *new_path = malloc(PATH_MAX);
+  int path_len = strlen(orig_path) + strlen(SNAP_INSTANCE_NAME) + 8;
+  char *new_path = malloc(path_len);
```
Comment (alexsander-souza) [inline snap-preload.c:32]: add a `assert(new_path != NULL)`. In an OoM scenario you want to crash giving some hint about the reason.

**Review by alexsander-souza — CHANGES_REQUESTED** [review]
a few comments inline

File: snap-preload.c (hunk 1 above)
Comment (albertodonato) [reply snap-preload.c]: fixed thanks

File: snap-preload.c (hunk 2 above)
Comment (albertodonato) [reply snap-preload.c]: fixed

**Review by alexsander-souza — APPROVED** [review]
+1

**Pull request merged**